nil/getting-started/essentials/handling-async-execution.mdx

```diff
@@ -0,0 +1,182 @@
+# Handling async execution
+
+To support [**cross-shard communication**](../../principles#cross-shard-communication), =nil; supports smart contracts making async calls to one another directly in Solidity code. 
+
+This tutorial provides a brief theoretical and practical primer on async execution, and gives suggestions on how this feature can be integrated into an application.
+
+## Performing an async call
+
+Consider a smart contract with the following code:
+
+```sol showLineNumbers
+import "./nil.sol";
+
+contract Caller {
+
+    using nil for address; 
+
+    event CallCompleted(address indexed dst)
+
+    function call(address dst) public payable {
+        dst.async_call(address(0), gasleft(), false, msg.value, abi.encodeWithSignature("funcName"));
+        emit CallCompleted(dst);
+    }
+}
+```
+
+The `address.async_call()` function makes it possible for the contract to call a function in another smart contract regardless of the shard where this contract is located. There are no additional actions required for making async calls: simply adding `address.async_call()` and specifying its arguments is sufficient.
+
+:::info
+
+The `address.async_call()` function is a shortcut for executing a precompiled contract responsible for managing async calls across the cluster.
+
+:::
+
+:::info[Async calls on the same shard]
+
+`address.async_call()` works even if the contract with the specified address is located on the same shard. The mechanism remains exactly the same as with cross-shard communications: the function will produce a new transaction, and the requested function in another contract will be executed whenever said transaction is processed by the shard. 
+
+:::
+
+## Receiving an async call
+
+The contract receiving [**an external message**](../../core-concepts/shards-parallel-execution##internal-vs-external-messages) must implement an additional function so that this call is processed. Unless this function is present, a contract cannot accept external messages. The function signature:
```

Comment:
If you're describing a way to send an async message (internal) in the section above, it's a bit confusing that you follow it up with "receiving an async call" where you describe external messages.

It would be better to have two sections - one for internal calls, and another for external calls.